### ISSUE TITLE
Migrate to RESTEasy Reactive

### ISSUE THREAD

**Issue by hbelmiro** [quarkiverse/quarkus-openapi-generator#150]
This extension uses RESTEasy Classic. However, Quarkus recommends using RESTEasy Reactive now.
Documentation: https://quarkus.io/guides/resteasy-reactive-migration

**Comment by melloware** [comment]
I would like a switch bewteen the two if possible because a lot of us still use RestEasy Classic.

**Comment by melloware** [comment]
Isn't this also similar to this ticket: https://github.com/quarkiverse/quarkus-openapi-generator/issues/82

Read that discussion where it talks about generating different client types.

**Comment by hbelmiro** [comment]
@melloware I don't think so. This is just to replace the RESTEasy classic dependency with the reactive one. It shouldn't result in significant changes.

**Comment by hbelmiro** [comment]
There's this compatibility break that affects Multipart requests: https://github.com/quarkusio/quarkus/issues/23059#issue-1109278469
We need a definition from Quarkus before proceeding with this.

**Comment by melloware** [comment]
Good I am heavily using MultiPart requests.

**Comment by ricardozanini** [comment]
Relates to #157

**Comment by einarjohnson** [comment]
Any ETA on when this might be fixed/become compatible?

**Comment by hbelmiro** [comment]
> Any ETA on when this might be fixed/become compatible?

@einarjohnson I'm working on this.
There's [this compatibility](https://github.com/quarkusio/quarkus/issues/23059#issuecomment-1287053067) break I still need to assess and there might have others that I'm still not aware of. Therefore I don't have a ETA. Sorry.

**Comment by einarjohnson** [comment]
@hbelmiro, thanks for getting back to me so quickly.

**Comment by ssaip** [comment]
Very much needed feature! Until this is done, we have to stick to the normal openapi-generator :(

**Comment by ricardozanini** [comment]
@ssaip unfortunately, the compatibility break, as documented in https://github.com/quarkusio/quarkus/issues/23059#issuecomment-1287053067 prevents us from migrating.

We can migrate, but we will lose a feature. We can hear from the community and people following this thread whether or not the benefit of using RestEasy reactive is ok to lose the feature.

@hbelmiro, or we can study a solution for users to pick their favorite. We will have to keep a few different templates, but worth having if someone is interested in contributing.

**Comment by hbelmiro** [comment]
@ricardozanini Yes. We had this discussion a few months ago. The idea is to support both implementations. The extension would figure out which one the user has on its dependencies and generate the code for that one.
I already started working on this, but it's on hold because of that compatibility break.
The idea of supporting reactive with caveats sounds good to me. We can add support to multipart in the future when that issue is fixed.

**Comment by hbelmiro** [comment]
Good news guys: https://github.com/quarkusio/quarkus/pull/30285

**Comment by maxandersen** [comment]
Just a comment here - do not just rely on approach based on what is on the classpath. You can use the reactive client in blocking mode just fine. Just like when using reactive resteasy server side by default we dont generate mutiny style apis but use classic. 

Thus I would say it should (at least) be possible to control with a "style" toggle.

**Comment by ricardozanini** [comment]
Thanks @maxandersen! @hbelmiro just started working on the migration, we should get news soon.

**Comment by chberger** [comment]
@ricardozanini Any update on this? We're waiting for this feature 👍

**Comment by ricardozanini** [comment]
Hi! Nice to see that you're waiting for this feature, but unfortunately the person that was working on it is on PTO. We should get updates by the end of the month. 

If it's super critical for you, you can send a PR and I can review it and do a release.

**Comment by hbelmiro** [comment]
> Just a comment here - do not just rely on approach based on what is on the classpath. You can use the reactive client in blocking mode just fine. Just like when using reactive resteasy server side by default we dont generate mutiny style apis but use classic.
> 
> Thus I would say it should (at least) be possible to control with a "style" toggle.

Sorry, @maxandersen. I'm not sure I understood your comment.

Rather than replacing the classic dependency with the reactive one, the idea is to leave it up to the application that uses quarkus-openapi-generator to decide on which implementation to use. And the generated code will be blocking no matter the dependency being used by the application.
Why? Because using the reactive dependency on this extension would break applications that use the classic one.
So why not just remove the dependency and generate blocking code not caring about what dependency the application is using, since it's supported by both dependencies? That's the idea, but although blocking code is supported by classic and reactive, the code has slight differences. So, we need to know what dependency the application uses to generate the blocking code correctly.

**Comment by chberger** [comment]
> Hi! Nice to see that you're waiting for this feature, but unfortunately the person that was working on it is on PTO. We should get updates by the end of the month.
> 
> If it's super critical for you, you can send a PR and I can review it and do a release.

here is my try https://github.com/quarkiverse/quarkus-openapi-generator/pull/263

**Comment by chberger** [comment]
Oh boy, now @hbelmiro implemented it after all: https://github.com/quarkiverse/quarkus-openapi-generator/pull/262. Too bad, I could have invested my time better ...

**Comment by ricardozanini** [comment]
@chberger I'm sorry to hear that. You can still review @hbelmiro's PR if you don't mind.

**Comment by chberger** [comment]
@ricardozanini Sure, I'd like to do that. However, I can have a look into it on Friday EOB at the earliest.

**Comment by ricardozanini** [comment]
No worries, @chberger. Do it as you feel like! Thanks for your contribution.

**Comment by chberger** [comment]
@ricardozanini Could you add me as a reviewer?

**Comment by ricardozanini** [comment]
@chberger I can't since you're not a member. You can do the review/comment even without it, don't worry.
